components/brave_extension/extension/brave_extension/background/greaselion.ts

```diff
@@ -393,3 +418,18 @@ chrome.runtime.onConnectExternal.addListener((port: chrome.runtime.Port) => {
     }
   })
 })
+
+chrome.runtime.onMessageExternal.addListener(
+  function (msg: any, sender: chrome.runtime.MessageSender, sendResponse: any) {
+    if (!msg) {
```

Comment:
Do we need to filter extensions here (e.g. filter for rewards extension or greaselion extensions), for security reasons?

Reply:
Do you mean similar to what we do in the `onConnectExternal` listener?

Reply:
Right - given that we have `externally_connectable` opened up, is there any way that some random extension could send a message through this listener?

Reply:
Will enter a follow-up issue to address this.

Reply:
This will absolutely be required and needs to be uplifted to wherever this PR is ASAP

Reply:
Agreed, will get it in tomorrow.